Vagrantfile

```diff
@@ -0,0 +1,40 @@
+Vagrant.configure("2") do |config|
+  config.vm.box = "hashicorp/bionic64"
+
+  config.vm.network "forwarded_port", guest: 5000, host: 5000
+
+  config.vm.provision "shell", privileged: false, inline: <<-SHELL
+
+    sudo apt update
+    sudo apt install -y build-essential libssl-dev zlib1g-dev \
+    libbz2-dev libedit-dev libsqlite3-dev wget curl llvm \
+    libncurses5-dev xz-utils tk-dev libxml2-dev libxmlsec1-dev libffi-dev liblzma-dev
+
+    # Install pyenv, a python version manager 
+    rm -rf ~/.pyenv
+    git clone https://github.com/pyenv/pyenv.git ~/.pyenv
+    echo 'export PYENV_ROOT="$HOME/.pyenv"' >> ~/.profile
+    echo 'export PATH="$PYENV_ROOT/bin:$PATH"' >> ~/.profile
+    echo 'eval "$(pyenv init --path)"' >> ~/.profile
+    source ~/.profile
+    
+    # Install Python 3.9.6 and make it the default python version
+    pyenv install 3.9.6
+    pyenv global 3.9.6
+
+    # Download and install the latest version of Poetry
+    curl -sSL https://raw.githubusercontent.com/python-poetry/poetry/master/get-poetry.py | python
+  
+  SHELL
+
+  config.trigger.after :up do |trigger|
+    trigger.name = "Launching App"
+    trigger.info = "Running the TODO app setup script"
+    trigger.run_remote = {privileged: false, inline: "
+      cd /vagrant
+      poetry install
+      poetry run flask run --host=0.0.0.0
```

Comment:
Slightly nicer to run `nohup poetry run flask run > app.log 2>&1 &` so that the "up" trigger completes (by starting Flask as a background process) and Vagrant can better manage the state of the VM.